### ISSUE TITLE
[Core] update vendored packages / use glide

### ISSUE THREAD

**Issue by zramsay** [eris-ltd/eris#929]

**Comment by pietv** [comment]
"use glide" for what? why?

**Comment by zramsay** [comment]
to version the dependencies in `/vendor`

**Comment by pietv** [comment]
Using an external tool to populate go dependencies will prevent CLI being installed with `go get` which I very much like to keep. In my opinion, the `glide` to tool is immature and very messy. 

Being certain of versions populated in `/vendor` can be made with pulling them from their respective repositories.

**Comment by zramsay** [comment]
we're going to need some form of dependency management for the issues in this epic; pm & cm moving into cli have dependencies on keys & edb for which we need to be cleanly managed.

The ideal solution discussed when we began the sprint was to have the dependencies checked-in to the repo, so that it remains `go get`-able but use `glide` for updating. @VoR0220 can you provide an overview for implementing glide in #1083

**Comment by VoR0220** [comment]
@pietv it won't prevent it from being go gettable, you can still do so just by checking in the vendor and that will preserve it's `go get`-ability. But yes, as for glide, it's really a matter of properly defining what you're using. So here's the current CLI glide yaml just based off of running a `glide init` in the repo on epm consolidate. 

```
package: github.com/eris-ltd/eris-cli
import:
- package: github.com/BurntSushi/toml
- package: github.com/bugsnag/bugsnag-go
- package: github.com/docker/docker
  subpackages:
  - pkg/jsonmessage
  - pkg/term
- package: github.com/eris-ltd/common
  subpackages:
  - go/docs
- package: github.com/eris-ltd/eris-compilers
  subpackages:
  - network
  - util
- package: github.com/eris-ltd/eris-db
  subpackages:
  - client
  - client/core
  - keys
  - txs
- package: github.com/eris-ltd/eris-keys
  subpackages:
  - crypto
  - eris-keys
- package: github.com/fsouza/go-dockerclient
  subpackages:
  - external/github.com/docker/docker/pkg/archive
- package: github.com/kr/text
  subpackages:
  - colwriter
- package: github.com/oleiade/reflections
- package: github.com/pborman/uuid
- package: github.com/rs/cors
- package: github.com/serenize/snaker
- package: github.com/spf13/cobra
- package: github.com/spf13/pflag
- package: github.com/spf13/viper
- package: github.com/tcnksm/go-gitconfig
- package: github.com/tendermint/go-wire
- package: golang.org/x/sys
  subpackages:
  - unix
- package: gopkg.in/yaml.v2
testImport:
- package: github.com/stretchr/testify
  subpackages:
  - assert
```

This basically implements all the packages and subpackages (packages of those required packages that are referenced). We can also specify specific git branches and repos as to where to import from like such

```
package: github.com/Masterminds/glide
import:
  - package: github.com/Masterminds/semver
  - package: github.com/Masterminds/cookoo
    version: ^1.2.0
    repo: git@github.com:Masterminds/cookoo.git
```

version can also be a commit hash, so we can lock them in based on the glide.yaml. So if there are any vendored dependencies that don't exactly "map" to their go gettable forms, those should be clearly defined in the glide.yaml file and specified as to where to get it from.

**Comment by zramsay** [comment]
closed by #1177
